agent/acs/handler/acs_handler.go

```diff
@@ -327,6 +312,110 @@ func startACSSession(ctx context.Context, client wsclient.ClientServer, timer tt
 	}
 }
 
+func (session *Session) computeReconnectDelay(isInactiveInstance bool) time.Duration {
+	if isInactiveInstance {
+		return session._inactiveInstanceReconnectDelay
+	}
+
+	return session.backoff.Duration()
+}
+
+func (session *Session) waitForDurationOrCancelledSession(delay time.Duration) bool {
```

Comment:
nit: I'm not wild about functions with "or" or "and" in the name, as it generally implies they do two things.  For this specific case, I think `waitForDuration` is probably fine and making the return `ok` or `done` conveys the meaning.